crux/models/resource.py

```diff
@@ -65,107 +47,114 @@ def __init__(
-        self._provenance = provenance
-        self._as_of = as_of
-        self._created_at = created_at
-        self._storage_id = storage_id
-        self._media_type = media_type
-        self._modified_at = modified_at
-        self._tags = tags
-        self._folder = None
-        self._labels = labels if labels else {}  # type: Dict[str, str]
-        self._folder = folder
 
+        self._folder = None
+        self.object_model = object_model
         self.connection = connection
-        self.raw_response = raw_response
 
     @property
     def id(self):
         """str: Gets the Resource ID."""
-        return self._id
+        return self.object_model["resourceId"]
 
     @property
     def description(self):
         """str: Gets the Resource Description."""
-        return self._description
+        return self.object_model["description"]
+
+    @description.setter
+    def description(self, description):
+        self.object_model["description"] = description
 
     @property
     def media_type(self):
         """str: Gets the Resource Description."""
-        return self._media_type
+        return self.object_model["mediaType"]
 
     @property
     def dataset_id(self):
         """str: Gets the Dataset ID."""
-        return self._dataset_id
+        return self.object_model["datasetId"]
 
     @property
     def folder_id(self):
         """str: Gets the Folder ID."""
-        return self._folder_id
+        return self.object_model["folderId"]
 
     @property
     def storage_id(self):
         """str: Gets the Storage ID."""
-        return self._storage_id
+        return self.object_model["storageId"]
 
     @property
     def name(self):
         """str: Gets the Resource Name."""
-        return self._name
+        return self.object_model["name"]
+
+    @name.setter
+    def name(self, name):
+        self.object_model["name"] = name
 
     @property
     def config(self):
         """str: Gets the config."""
-        return self._config
+        return self.object_model["config"]
+
+    @config.setter
+    def config(self, config):
+        self.object_model["config"] = config
 
     @property
     def provenance(self):
         """str: Gets the Provenance."""
-        return self._provenance
+        return self.object_model["provenance"]
+
+    @provenance.setter
+    def provenance(self, provenance):
+        self.object_model["provenance"] = provenance
 
     @property
     def type(self):
         """str: Gets the Resource Type."""
-        return self._type
+        return self.object_model["type"]
 
     @property
     def tags(self):
         """:obj:`list` of :obj:`str`: Gets the Resource Tags."""
-        return self._tags
+        return self.object_model["tags"]
+
+    @tags.setter
+    def tags(self, tags):
+        self.object_model["tags"] = tags
 
     @property
     def labels(self):
         """dict: Gets the Resource labels."""
-        return self._labels
+        labels = {}
```

Comment:
What would happen if a user does this:

```python
file.labels["k1"] = "v1"
file.labels["k1"] = "v1"
file.labels.update({"k2": "v2"})
```

I'm thinking we need to handle that. Like maybe we need a special `LabelMapping` object that allows people to work with it like a `dict` then maps it to the correct API representation. Where `object_model` would have the correct API representation, but `file.label` would be a `LabelMapping` the gets and sets from `object_model` while exposing a `dict` interface.